src/cascadia/TerminalApp/ActionArgs.idl

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace TerminalApp
+{
+    // An empty interface must specify an explicit [uuid] to ensure uniqueness.
+    // We also manually have to specify a "version" attribute to make the compiler happy.
+    [uuid("191C2BDE-1A60-4BAB-9765-D850F0EF2CAC")][version(1)] interface IActionArgs{};
+
+    interface IActionEventArgs
+    {
+        Boolean Handled;
+        IActionArgs ActionArgs;
+    };
+
+    enum Direction
+    {
+        Left = 0,
+        Right,
+        Up,
+        Down
+    };
+
+    [default_interface] runtimeclass ActionEventArgs : IActionEventArgs
+    {
+        ActionEventArgs(IActionArgs args);
+    };
+
+    [default_interface] runtimeclass CopyTextArgs : IActionArgs
+    {
+        Boolean TrimWhitespace;

```

Comment:
I wonder. Should these all be `{get;}`? I do not want a recipient of a static instance of one of these to be able to, even accidentally, change it.

Reply:
this would mandate that they are _set_ only by a private impl